### ISSUE TITLE
cleanup "use local variable type interference" refactors generics and removes needed import

### ISSUE THREAD

**Issue by carstenartur** [eclipse-jdt/eclipse.jdt.ui#1525]
eclipse.buildId=4.33.0.20240711-1200
java.version=22.0.1
java.vendor=Eclipse Adoptium
BootLoader constants: OS=macosx, ARCH=x86_64, WS=cocoa, NL=de_DE

Code like this:

```java
HashSet<IJavaProject> selectedElements= new HashSet<>();
```

is refactored to 

```java
var selectedElements= new HashSet<IJavaProject>();
```

Unfortunately the resulting code has the error "IJavaProject cannot be resolved to a type" because the import is removed.

**Comment by carstenartur** [comment]
Can it be that the method org.eclipse.jdt.internal.corext.refactoring.structure.ImportRemover.divideTypeRefs() simply is not smart enough to handle parameterized types?

**Comment by jjohnstn** [comment]
Closed with #1538 merged.
